### PR TITLE
feat: set permissions and member field optional in custom-role sub-module

### LINKED ISSUE #194: Submodule custom_roles_iam change permission and member field to optional
### TL;DR

When user defines a role using `base_roles` field, there is actually no need to have a redundant `permission` field. Also, there is no need to force user to add `member`, when creating a role

### Terraform Resources

_No response_

### Detailed design

Just need to simply add `default = []` in `variables.tf` to these two variables I mentioned above and problem solved.
```markdown
variable "permissions" {
  type        = list(string)
  description = "IAM permissions assigned to Custom Role."
  default     = []
}

variable "members" {
  description = "List of members to be added to custom role."
  type        = list(string)
  default     = []
}

```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by imrannayer**
Fix #194

### CODE REVIEW

File: modules/custom_role_iam/variables.tf [hunk 1]
```diff
@@ -27,13 +27,14 @@ variable "title" {
 
 variable "base_roles" {
   type        = list(string)
-  description = "List of base predefined roles to use to compose custom role."
+  description = "List of base predefined roles to use to compose custom role. Either base_roles or permissions must be set with some values"
   default     = []
```
Comment (g-awmalik) [inline modules/custom_role_iam/variables.tf:31]: Can we add a validation check here and to permissions to confirm if either variable is set?

**Review by g-awmalik — CHANGES_REQUESTED** [review]
@imrannayer - PTAL at my comment if its reasonable.

**Comment by imrannayer** [comment]
> @imrannayer - PTAL at my comment if its reasonable.

@g-awmalik cross variable validation is not supported in Terraform yet. I can add a check locally and only allow resource creation when either one of them is non-empty list but that will not make it clear to the user why resources are not creating.

**Comment by g-awmalik** [comment]
I see. We can let go this for now but it'll be nice if this can be made into an object instead with two attributes. That way we can check if at least one attribute is set.

**Review by g-awmalik — APPROVED** [review]

**Comment by imrannayer** [comment]
@g-awmalik can you plz merge. I dont have permission on this repo.

**Pull request merged**